### ISSUE TITLE
Would be convenient if console.log wrote to Android's LOGCAT

### ISSUE THREAD

**Issue by jackpal** [paddybyers/anode#9]
It would be convenient if console.log and related functions wrote to Android's LOGCAT by default.

See android/log.h in the NDK.

**Comment by paddybyers** [comment]
Yes, I've been meaning to do this for a while. Obviously a pull request would be great :)

**Comment by jackpal** [comment]
Hmm, I see the problem is that console writes to stdout, which Android black-holes.

So maybe it would be better to provide a separate android log API, and maybe a wrapper that's compatible with console.

How would you go about implementing such a thing?

One way would be to export the Android NDK log APIs as a new node module.

Another way would be to export the Android Java log APIs as a new node module.

Is either of that supported for anode? Are there any examples of Android APIs being exported?

Or is there another way?

**Comment by NomadLibra** [comment]
You can redirect stdio to logcat.

$ adb shell stop
$ adb shell setprop log.redirect-stdio true
$ adb shell start

**Comment by jackpal** [comment]
Thanks!
On Jul 10, 2012 12:04 PM, "NomadLibra" <
reply@reply.github.com>
wrote:
